### PR TITLE
#162 MSAL should return an error when no access_token is provided

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: MSAL/test/unit/MSALInteractiveRequestTests.m [hunk 1]
```diff
@@ -707,4 +707,150 @@ - (void)testInteractiveRequestFlowWithUserNotMatch
...
+    
+    __block NSUUID *correlationId = [NSUUID new];
+    
+    MSALRequestParameters *parameters = [MSALRequestParameters new];
+    parameters.urlSession = [MSALTestURLSession createMockSession];
+    parameters.scopes = [NSOrderedSet orderedSetWithArray:@[@"fakescope1", @"fakescope2"]];
+    parameters.unvalidatedAuthority = [NSURL URLWithString:@"https://login.microsoftonline.com/common"];
+    parameters.redirectUri = [NSURL URLWithString:UNIT_TEST_DEFAULT_REDIRECT_URI];
+    parameters.clientId = UNIT_TEST_CLIENT_ID;
+    parameters.extraQueryParameters = @{ @"eqp1" : @"val1", @"eqp2" : @"val2" };
```
Comment (unpluggedk) [inline MSAL/test/unit/MSALInteractiveRequestTests.m]: Do we need this to be checked? We check this in different test.

File: MSAL/test/unit/MSALInteractiveRequestTests.m [hunk 2]
```diff
@@ -707,4 +707,150 @@ - (void)testInteractiveRequestFlowWithUserNotMatch
...
+           @"eqp2" : @"val2",
+           @"redirect_uri" : UNIT_TEST_DEFAULT_REDIRECT_URI,
+           @"response_type" : @"code",
+           @"code_challenge": pkce.codeChallenge,
+           @"code_challenge_method" : @"S256",
+           @"uid" : @"true",
+           @"slice" : @"testslice"
+           };
+         NSDictionary *QPs = [NSDictionary msalURLFormDecode:url.query];
+         XCTAssertTrue([expectedQPs compareToActual:QPs]);
```
Comment (unpluggedk) [inline MSAL/test/unit/MSALInteractiveRequestTests.m]: related to above comment, we could prob skip this

Comment (oldalton) [inline MSAL/test/unit/MSALInteractiveRequestTests.m]: yes, I was also thinking that tests in here are very repetitive and can probably be improved.

File: MSAL/src/MSALError_Internal.m [hunk 3]
```diff
@@ -47,6 +47,7 @@
         STRING_CASE(MSALErrorInvalidResponse);
         STRING_CASE(MSALErrorBadAuthorizationResponse);
         STRING_CASE(MSALErrorAuthorizationFailed);
+        STRING_CASE(MSALErrorNoAccessTokeInResponse);
```
Comment (RPangrle) [inline MSAL/src/MSALError_Internal.m]: nit: Spelling

File: MSAL/src/public/MSALError.h [hunk 4]
```diff
@@ -100,6 +100,12 @@ typedef NS_ENUM(NSInteger, MSALErrorCode)
     MSALErrorAuthorizationFailed = -42104,
     
     /*!
+        MSAL received a valid token response, but it didn't contain an access token.
+        By Oauth 2.0 spec it should never happen, but it still occasionally happens.
```
Comment (RPangrle) [inline MSAL/src/public/MSALError.h]: Replace "but it still occasionally happens" with, "check to make sure your application is consented get all of the scopes you are asking for."

File: MSAL/src/MSALError_Internal.m (hunk 3 above)
Comment (oldalton) [inline MSAL/src/MSALError_Internal.m]: Fixed

**Pull request merged**